src/accel/detail/IntegrationSingleton.hh

```diff
@@ -91,6 +99,10 @@ class IntegrationSingleton
     real_type stop_timer() { return get_time_(); }
 
   private:
+    //// TYPES ////
+
+    using VariantOffload = std::variant<LocalTransporter, LocalOpticalOffload>;
```

Comment:
It's a little weird to have a variant of two virtual classes that share the same interface. And even weirder to have an optional of that variant (usually you'd use `variant<std::monostate, ...>`.

Why not `std::unique_ptr<LocalOffloadInterface>` plus dynamic-cast?